doc/source/examples/03_serving_stable_diffusion/app.py

```diff
@@ -0,0 +1,73 @@
+from io import BytesIO
+
+import ray
+from ray import serve
+from fastapi import FastAPI
+from fastapi.responses import Response
+
+
+NUM_NODES = len(ray.nodes())
+
+app = FastAPI()
+
+
+@serve.deployment(num_replicas=1, route_prefix="/")
+@serve.ingress(app)
+class APIIngress:
+    def __init__(self, diffusion_model_handle) -> None:
+        self.handle = diffusion_model_handle
+
+    @app.get(
+        "/imagine",
+        responses={200: {"content": {"image/png": {}}}},
+        response_class=Response,
+    )
+    async def generate(self, prompt: str, img_size: int = 512):
+        assert len(prompt), "prompt parameter cannot be empty"
+
+        image = await (await self.handle.generate.remote(prompt, img_size=img_size))
+
+        file_stream = BytesIO()
+        image.save(file_stream, "PNG")
+        return Response(content=file_stream.getvalue(), media_type="image/png")
+
+
+def get_runtime_env():
```

Comment:
Don't use runtime envs.

Reply:
I ended up using `runtime_env` due to `pip install --user` being really slow (due to EFS) to install the packages: 6 minutes compared to 1 minute for a regular `pip install`.

`runtime_env` is actually much faster to setup dependencies (it takes ~2 minutes to get the runtime env dependencies installed + get model pulled from huggingface).

Richard recommended this to get around introducing the runtime env concept:

> What if we don’t use the word runtime env, and instead used the wording of
“To start, we’ll set up our dependencies by passing this configuration to Ray. Ray will install all packages in this requirements list on each node.
There are other ways of installing dependencies that you can take a look at here (docs)”
ray.init(runtime_env={"pip": "./requirements.txt")

I'm okay with going with `pip install --user` if this initial buffer time of 5-8 minutes is not a dealbreaker. Personally, I've found it very nice to be able to run the script right away, without needing to install anything myself.

Reply:
How about this instead?
1. We make the default config to use a single GPU node for the workspace.
2. Then, users can use normal `pip install` which is very fast. We can say use `pip install --user` or runtime_envs if you want to scale to a cluster.

Reply:
Sounds good, let's go with this!